halo2-base/src/gates/range.rs

```diff
@@ -316,6 +383,10 @@ pub trait RangeInstructions<F: ScalarField> {
         (div, rem)
     }
 
+    /// Returns the last bit of the [a] of a 2^64 bit number
```

Comment:
We didn't explain this well (and this function is kind of broken right now), but `a` can be larger than `u64`. We just assume it has been range checked to `limb_bits` bits (this could be `200` or something)